apis/python/src/tiledb/vector_search/index.py

```diff
@@ -185,7 +188,9 @@ def query(self, queries: np.ndarray, k, **kwargs):
                 if (
                     internal_results_d[query_id, res_id] == 0
                     and internal_results_i[query_id, res_id] == 0
+                    and not self.is_type_erased_index()
                 ):
+                    # NOTE(paris): This is skipped for type-erased modules because is is valid to have an ID of 0 with a score of 0.
                     internal_results_d[query_id, res_id] = MAX_FLOAT_32
                     internal_results_i[query_id, res_id] = MAX_UINT64
```

Comment:
I  disabled this check which will throw out query results if the ID is 0 and the score is 0. @NikolaosPapailiou or @lums658 , do you know why we had it? Should we perhaps also remove it for non-type-erased code?
```
if (internal_results_d[query_id, res_id] == 0 and internal_results_i[query_id, res_id] == 0):
    internal_results_d[query_id, res_id] = MAX_FLOAT_32
    internal_results_i[query_id, res_id] = MAX_UINT64
```

Reply:
The convention for all our query implementations is that empty results should be represented as `MAX_UINT64, MAX_FLOAT_32` I think we should be consistent here and fix it on the C++ side rather than patching it here

Reply:
Yes, that's why I removed the check here. I don't understand why we have it because in the type-erased code we use the convention you describe. 

I was just asking if I could also remove it for the non-type-erased case, i.e. if you knew why we had this check?

Reply:
Thanks sorry for the misunderstanding, yes at some point our queries returned 0 for empty results. This was added for backwards compatibility. I think we can remove it now

Reply:
Sounds good, I'll remove it in a follow-up to keep the changes self-contained.

Reply:
Removed here: https://github.com/TileDB-Inc/TileDB-Vector-Search/pull/319